functions/src/index.ts

```diff
@@ -166,3 +169,98 @@ export const loadFirestore = functions.https.onRequest(
     response.status(200).send();
   }
 );
+
+/*
+ * Listen for changes to the User document
+ * and initiate the appropriate Glean ping(s)
+ */
+export const handleUserChangesImpl = async function (
+  change: Change<DocumentSnapshot>,
+  context: EventContext
+): Promise<Boolean> {
+  const userID = context.params.userID;
+  // Get an object with the current document value.
+  // If the document does not exist, it has been deleted.
+  const newUser = change.after.exists ? change.after.data() : null;
+
+  // Get the old document, to compare the enrollment state.
+  const oldUser = change.before.exists ? change.before.data() : null;
+
+  if (!newUser || (oldUser && oldUser.enrolled === true && !newUser.enrolled)) {
+    // User document has been deleted
+    functions.logger.info(
+      `Sending deletion and unenrollment pings for user ID ${userID}`
+    );
+    // TODO send Glean pings
+    return true;
+  }
+
+  if ((!oldUser || !oldUser.enrolled) && newUser.enrolled === true) {
+    // User just enrolled
+    functions.logger.info(`Sending enrollment ping for user ID ${userID}`);
+    // TODO send Glean ping
```

Comment:
@thomik-corp  I wait on returning true in case the Firebase document change for enrollment and demographics update gets lumped together in the same change. This way we don't return prematurely and miss out on sending the Glean ping for the demographic change.

It's an unlikely scenario but I didn't see any harm in leaving it this way. Let me know if you feel otherwise.